### PR TITLE
Add alpha numeric mode support

### LINKED ISSUE #33: support of modeAlphaNum
do you plan to implement modeAlphaNum? For standard urls it uses less space and makes the qr-codes smaller...
best,
stefano

### PR DESCRIPTION
**Pull request by jtmcdole**
* QrByte interface extracted to QrDatum
* QrByte, QrNumeric now reference implement this type
* New QrAlphaNumeric created

Fixes #33

### CODE REVIEW

**Comment by jtmcdole** [comment]
Rebasing and fixing

**Review by jtmcdole — COMMENTED** [review]
Rebased; minor version bump for new functionality.

File: pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: qr
-version: 3.0.0
+version: 3.1.0
```
Comment (kevmoo) [inline pubspec.yaml:2]: just 3.0.0 – we haven't published v3 yet

**Review by kevmoo — APPROVED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (jtmcdole) [reply pubspec.yaml]: Done!

**Pull request merged**